cli_tools/google-osconfig-agent/lock_linux.go

```diff
@@ -0,0 +1,53 @@
+//  Copyright 2018 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package main
+
+import (
+	"os"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/logger"
+)
+
+func obtainLock() {
+	lockFile := "/run/lock/osconfig_agent.lock"
+
+	err := os.MkdirAll(filepath.Dir(lockFile), 0755)
+	if err != nil && !os.IsExist(err) {
+		logger.Fatalf("Cannot obtain agent lock: %v", err)
+	}
+
+	f, err := os.OpenFile(lockFile, os.O_RDWR|os.O_CREATE, 0600)
+	if err != nil && !os.IsExist(err) {
+		logger.Fatalf("Cannot obtain agent lock: %v", err)
```

Comment:
minor: you might want to differentiate this message from the previous one so you know where it failed if you see it.

Reply:
The error contains line number and the actual error received.